tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9761,192 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+						log.FailOnError(err, "error getting drive size for pool [%s]", poolListForOps[i].Uuid)
+						//adding 4 drives to each pool which is having one drive.so that max drive per node is accomdated
+						for j := 1; j <= 4; j++ {
+							driveSize := drvSize * uint64(j)
+							expectedSize := (poolListForOps[i].TotalSize / units.GiB) + driveSize
+							isjournal, err := isJournalEnabled()
+							log.FailOnError(err, "Failed to check is journal enabled")
+							expectedSizeWithJournal := expectedSize
+							if isjournal {
+								journalSize := 3 * uint64(j)
+								expectedSizeWithJournal = expectedSizeWithJournal - journalSize
+							}
+							err = Inst().V.ExpandPool(poolListForOps[i].Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+							log.FailOnError(err, "error while expanding pool")
+							maxDriveAllowed = maxDriveAllowed - 1
+							resizeErr := waitForPoolToBeResized(expectedSize, poolListForOps[i].Uuid, isjournal)
+							dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+							err = Inst().V.RefreshDriverEndpoints()
+							log.FailOnError(err, "error refreshing volume endpoints")
+						}
+					}
+				}
+			}
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+			sNode, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "fialed while getting updated node pool list")
+			poolListOnNode, err := GetPoolsDetailsOnNode(sNode)
+			for i := 0; i < len(poolListOnNode); i++ {
+				drvMap, err := Inst().V.GetPoolDrives(&sNode)
+				log.FailOnError(err, "error getting pool drives from node [%s]", sNode.Name)
+				drvs := drvMap[fmt.Sprintf("%d", poolListOnNode[i].ID)]
+				if (POOL_MAX_CLOUD_DRIVES - len(drvs)) > 0 {
+					selectedPools = poolListOnNode[i].Uuid
+				}
+			}
+			if selectedPools == "" {
+				err := fmt.Errorf("failed to get a pool to add a drive")
+				log.FailOnError(err, "failed to get a pool to add a drive")
+			}
+			//add one more extra drive than allowed per node
+			poolToBeResized, err := GetStoragePoolByUUID(selectedPools)
+			log.FailOnError(err, "failed to get a pool to add a drive")
+			drvSize, err := getPoolDiskSize(poolToBeResized)
+			log.FailOnError(err, "error getting drive size for pool [%s]", selectedPools)
+			expectedSize := (poolToBeResized.TotalSize / units.GiB) + drvSize
+			err = Inst().V.ExpandPool(poolToBeResized.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+			dash.VerifyFatal(err != nil, false, fmt.Sprintf("error in pool expand with err %s", err))
+			poolStatus, err := getPoolLastOperation(poolToBeResized.Uuid)
+			if poolStatus.Status != api.SdkStoragePool_OPERATION_FAILED {
+				log.FailOnError(err, "error getting drive size for pool [%s]", selectedPools)
+				err = Inst().V.ExpandPool(poolToBeResized.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
```

Comment:
can you explain the logic why are you expanding pool again?

Reply:
as discussed will explain it to you